docs/releases/3.0.txt

```diff
@@ -66,6 +66,18 @@ async code before, this may trigger if you were doing it incorrectly. If you
 see a ``SynchronousOnlyOperation`` error, then closely examine your code and
 move any database operations to be in a synchronous child thread.
 
+Cross-DB JSONField
+------------------
+
+Django now includes a :class:`~django.db.models.JSONField` that works on all
+supported database backends. It features support for custom JSON encoder and
+decoder, along with lookups and transforms supported in previously
+:class:`PostgreSQL-only JSONField <django.contrib.postgres.fields.JSONField>`.
+
+If your project uses the PostgreSQL JSONField, it is advised to migrate and
+use the new JSONField. For now, the old JSONField is left as a reference to
+the new one and may be removed in future releases.
```

Comment:
This isn't currently true... I don't see the old code removed in favour of backwards compatibility imports from the new location. This should be done. Are we going to implement deprecation of the old field and lookups for this release?

Reply:
Ah yes, sorry about that. I hadn't done that because @felixxm said that a [workaround should probably be added to the migrations](https://github.com/django/django/pull/11452#issuecomment-508051077). I'm not sure how to do that, though. Is it by modifying `contrib.postgres.apps`, this line?
```python
3802: 'django.contrib.postgres.fields.JSONField',
```

Reply:
I think we should import newly defined objects from `django.db.models` in `django.contrib.postgres`. If do that for `JSONField` we'll want to make sure to override `deconstruct` on the shim to keep returning `'django.contrib.postgres.JSONField'` else it'll generate new spurious migrations.

Reply:
I have imported that. Is overriding `deconstruct` enough?

Reply:
> `3802: 'django.contrib.postgres.fields.JSONField',`

I believe that line is related to introspection, not migrations - that should be solved in the way that @charettes described. We should probably change this to point to the new field as we want to encourage that going forward.

Further, if we haven't already, we should update the documentation for the `JSONField` in `django.contrib.postgres` to be a small entry pointing to the new field with an admonition that developers should use the new one.

Reply:
Done.